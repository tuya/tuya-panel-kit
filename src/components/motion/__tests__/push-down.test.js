import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import Motion from '../index';

describe('Motion Render', () => {
  it('basic render', () => {
    const component = renderer
      .create(
        <Motion.PushDown
          show={true}
          onHide={jest.fn()}
          style={{ position: 'absolute', bottom: 0 }}
          dropHeight={200}
        >
          <View style={{ width: 120, height: 48 }} />
        </Motion.PushDown>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('inVisible render', () => {
    const component = renderer
      .create(
        <Motion.PushDown
          show={false}
          onShow={jest.fn()}
          style={{ position: 'absolute', bottom: 0 }}
          dropHeight={200}
        >
          <View style={{ width: 120, height: 48 }} />
        </Motion.PushDown>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('onHide render', () => {
    const onHide = jest.fn();
    jest.useFakeTimers();
    const component = renderer.create(
      <Motion.PushDown
        show={true}
        onHide={onHide}
        style={{ position: 'absolute', bottom: 0 }}
        dropHeight={200}
      >
        <View style={{ width: 120, height: 48 }} />
      </Motion.PushDown>
    );
    expect(component).toMatchSnapshot();
    component.update(
      <Motion.PushDown
        show={false}
        onHide={onHide}
        style={{ position: 'absolute', bottom: 0 }}
        dropHeight={200}
      >
        <View style={{ width: 120, height: 48 }} />
      </Motion.PushDown>
    );
    jest.runAllTimers();
    expect(onHide).toHaveBeenCalled();
    component.unmount();
    expect(component).toMatchSnapshot();
  });

  it('onShow render', () => {
    const onShow = jest.fn();
    jest.useFakeTimers();
    const component = renderer.create(
      <Motion.PushDown
        show={false}
        onShow={onShow}
        style={{ position: 'absolute', bottom: 0 }}
        dropHeight={200}
      >
        <View style={{ width: 120, height: 48 }} />
      </Motion.PushDown>
    );
    expect(component).toMatchSnapshot();
    component.update(
      <Motion.PushDown
        show={true}
        onShow={onShow}
        style={{ position: 'absolute', bottom: 0 }}
        dropHeight={200}
      >
        <View style={{ width: 120, height: 48 }} />
      </Motion.PushDown>
    );
    jest.runAllTimers();
    expect(onShow).toHaveBeenCalled();
    component.unmount();
    expect(component).toMatchSnapshot();
  });
});
