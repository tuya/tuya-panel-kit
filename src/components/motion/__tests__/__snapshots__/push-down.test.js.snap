// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Motion Render basic render 1`] = `
<View
  accessibilityLabel="PushDown"
  renderToHardwareTextureAndroid={false}
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "height": 48,
        "width": 120,
      }
    }
  />
</View>
`;

exports[`Motion Render inVisible render 1`] = `null`;

exports[`Motion Render onHide render 1`] = `
<View
  accessibilityLabel="PushDown"
  renderToHardwareTextureAndroid={false}
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "height": 48,
        "width": 120,
      }
    }
  />
</View>
`;

exports[`Motion Render onHide render 2`] = `null`;

exports[`Motion Render onShow render 1`] = `null`;

exports[`Motion Render onShow render 2`] = `null`;
